tests/conftest.py

```diff
@@ -730,6 +731,20 @@ def ganache(
     yield from _launch_image(image, timeout=timeout, max_attempts=max_attempts)
 
 
+@pytest.mark.integration
+@pytest.fixture(scope="function")
```

Comment:
Also, as @marcofavorito pointed out, this scope will be costly to setup and tear down on every test that uses this fixture, but on the other hand the whole point of having introduced this is to have clean separate image instances on different tests.

Reply:
true. Let's keep it with `function` for now; another approach could be to have a decorator for test functions that, in case of failure, would restart the SOEF and repeat the test.

Reply:
I think scope to class would be preferable no?

Reply:
We can do class and later change if need to.